### PR TITLE
gtkcord4: 0.0.12 -> 0.0.16-1

### PR DESCRIPTION
**Pull request by hmenke**
## Description of changes

New upstream releases:
- https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.13 [`v0.0.12...v0.0.13`](https://github.com/diamondburned/gtkcord4/compare/v0.0.12...v0.0.13)
- https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.14 [`v0.0.13...v0.0.14`](https://github.com/diamondburned/gtkcord4/compare/v0.0.13...v0.0.14)
- https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.15 [`v0.0.14...v0.0.15`](https://github.com/diamondburned/gtkcord4/compare/v0.0.14...v0.0.15)
- https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.15-1 [`v0.0.15...v0.0.15-1`](https://github.com/diamondburned/gtkcord4/compare/v0.0.15...v0.0.15-1)
- https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.16 [`v0.0.15-1...v0.0.16`](https://github.com/diamondburned/gtkcord4/compare/v0.0.15-1...v0.0.16)
- https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.16-1 [`v0.0.16...v0.0.16-1`](https://github.com/diamondburned/gtkcord4/compare/v0.0.16...v0.0.16-1)

AppID has been renamed: https://github.com/diamondburned/gtkcord4/issues/156

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/gtkcord4/default.nix [hunk 1]
```diff
@@ -18,13 +18,13 @@
 
 buildGoModule rec {
   pname = "gtkcord4";
-  version = "0.0.12";
+  version = "0.0.14";
```
Comment (FliegendeWurst) [inline pkgs/applications/audio/gtkcord4/default.nix:21]: https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.15-1 has been released

**Review by FliegendeWurst — APPROVED** [review]
Builds and launches to a login screen.

Result of `nixpkgs-review pr 259394` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gtkcord4</li>
  </ul>
</details>

File: pkgs/applications/audio/gtkcord4/default.nix [hunk 2]
```diff
@@ -52,12 +52,11 @@ buildGoModule rec {
...
-    install -D -m 444 internal/icons/png/logo.png $out/share/icons/hicolor/256x256/apps/gtkcord4.png
+    install -D -m 444 -t $out/share/applications nix/so.libdb.gtkcord4.desktop
+    install -D -m 444 internal/icons/hicolor/scalable/apps/logo.svg $out/share/icons/hicolor/scalable/apps/gtkcord4.svg
   '';
 
-  vendorHash = "sha256-LCLZBcYiexffvCr4vdZdIwNKo0s4mqPc6KxRumRhf1Y=";
+  vendorHash = "sha256-OoJA87jorUO/Qp3IuOMrTNi1GOhv+tiVU02r6uV79PI=";
 
   meta = with lib; {
     description = "GTK4 Discord client in Go, attempt #4.";
```
[the remark this answers is not in the document]
Comment (hmenke) [reply pkgs/applications/audio/gtkcord4/default.nix:62]: Would it be okay to also take that chance to move it to `pkgs/by-name`?

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/applications/audio/gtkcord4/default.nix:62]: > Would it be okay to also take that chance to move it to `pkgs/by-name`?

No, you don't have to do that.

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/applications/audio/gtkcord4/default.nix:62]: It's my fault, but I would appreciate it if you correct them in this pr.

[the remark this answers is not in the document]
Comment (FliegendeWurst) [reply pkgs/applications/audio/gtkcord4/default.nix]: > > Would it be okay to also take that chance to move it to `pkgs/by-name`?
> 
> No, you don't have to do that.

As I understand it, that will be done automatically eventually: https://github.com/NixOS/nixpkgs/pull/211832

**Review by FliegendeWurst — APPROVED** [review]

**Comment by diamondburned** [comment]
Quick note: v0.0.16 is out now, https://github.com/diamondburned/gtkcord4/releases/tag/v0.0.16

**Comment by hmenke** [comment]
@diamondburned Vendoring is currently broken with v0.0.16, but I don't know in which dependency. Is there a way to find that out?

```
go: downloading github.com/diamondburned/gotkit v0.0.0-20231117094938-ace9cfa5e044
go: downloading github.com/diamondburned/adaptive v0.0.2-0.20221227093656-fa139be203a8
go: downloading github.com/diamondburned/arikawa/v3 v3.3.1-0.20230609054540-4224b93c4e01
go: downloading github.com/diamondburned/chatkit v0.0.0-20231117102020-f172e4a1e35d
go: downloading github.com/diamondburned/gotk4/pkg v0.0.6-0.20231117083507-eb41da62b149
go: downloading github.com/diamondburned/ningen/v3 v3.0.1-0.20231114075203-ff2cd6b90304
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/diamondburned/gotk4-adwaita/pkg v0.0.0-20231117093505-26f1789529b0
go: downloading github.com/ianlancetaylor/cgosymbolizer v0.0.0-20220405231054-a1ae3e4bba26
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/enescakir/emoji v1.0.0
go: downloading github.com/sahilm/fuzzy v0.1.0
go: downloading github.com/yuin/goldmark v1.5.6
go: downloading golang.org/x/image v0.0.0-20220902085622-e7cb96979f69
go: downloading golang.org/x/sync v0.0.0-20220722155255-886fb9371eb4
go: downloading golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading go4.org v0.0.0-20200411211856-f5505b9728dd
go: downloading github.com/leonelquinteros/gotext v1.5.3-0.20230829162019-37f474cfb069
go: downloading github.com/yalue/merged_fs v1.2.3
go: downloading github.com/alecthomas/chroma v0.10.0
go: downloading github.com/zalando/go-keyring v0.2.1
go: downloading golang.org/x/crypto v0.1.0
go: downloading github.com/gorilla/schema v1.2.0
go: downloading github.com/twmb/murmur3 v1.1.3
go: downloading go4.org/unsafe/assume-no-moving-gc v0.0.0-20230525183740-e7c30c78aeb2
go: downloading golang.org/x/text v0.4.0
go: downloading github.com/alessio/shellescape v1.4.1
go: downloading github.com/danieljoos/wincred v1.1.0
go: downloading github.com/godbus/dbus/v5 v5.0.6
go: downloading github.com/dlclark/regexp2 v1.4.0
go: pattern codeblock_test_want.txt: no matching files found
```

**Comment by diamondburned** [comment]
Whoops. This should be fixed in v0.0.16-1.

**Comment by hmenke** [comment]
Adding the backport label because I don't want 23.11 users to be stuck on an ancient and outdated version of gtkcord4.

**Pull request merged**